### PR TITLE
cpu/stm32/wl: initial periph_adc implementation

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR adds an initial implementation for `periph_adc` on the stm32wl plarforms.

### Testing procedure

Run `tests/periph_adc`, here had the pins plugged into GND and then switched to 3.3V (note that there are some suppurios measurements when pi is floating)

```
2021-11-11 14:29:12,174 # ADC_LINE(1): 1
2021-11-11 14:29:12,273 # ADC_LINE(0): 1
2021-11-11 14:29:12,274 # ADC_LINE(1): 0
2021-11-11 14:29:12,373 # ADC_LINE(0): 1
2021-11-11 14:29:12,374 # ADC_LINE(1): 1
2021-11-11 14:29:12,474 # ADC_LINE(0): 1
2021-11-11 14:29:12,474 # ADC_LINE(1): 2
2021-11-11 14:29:12,573 # ADC_LINE(0): 3
2021-11-11 14:29:12,574 # ADC_LINE(1): 1
2021-11-11 14:29:12,673 # ADC_LINE(0): 0
2021-11-11 14:29:12,674 # ADC_LINE(1): 2
2021-11-11 14:29:12,774 # ADC_LINE(0): 0
2021-11-11 14:29:12,774 # ADC_LINE(1): 2
2021-11-11 14:29:12,874 # ADC_LINE(0): 0
2021-11-11 14:29:12,874 # ADC_LINE(1): 1
2021-11-11 14:29:12,973 # ADC_LINE(0): 3
2021-11-11 14:29:12,974 # ADC_LINE(1): 1
2021-11-11 14:29:13,074 # ADC_LINE(0): 3
2021-11-11 14:29:13,074 # ADC_LINE(1): 602
2021-11-11 14:29:13,174 # ADC_LINE(0): 1
2021-11-11 14:29:13,174 # ADC_LINE(1): 2808
2021-11-11 14:29:13,274 # ADC_LINE(0): 15
2021-11-11 14:29:13,275 # ADC_LINE(1): 2801
2021-11-11 14:29:13,374 # ADC_LINE(0): 3
2021-11-11 14:29:13,374 # ADC_LINE(1): 2811
2021-11-11 14:29:13,474 # ADC_LINE(0): 49
2021-11-11 14:29:13,475 # ADC_LINE(1): 2805
2021-11-11 14:29:13,574 # ADC_LINE(0): 5
2021-11-11 14:29:13,574 # ADC_LINE(1): 2809
2021-11-11 14:29:13,674 # ADC_LINE(0): 2
2021-11-11 14:29:13,674 # ADC_LINE(1): 2805
2021-11-11 14:29:13,774 # ADC_LINE(0): 3
2021-11-11 14:29:13,774 # ADC_LINE(1): 2818
2021-11-11 14:29:13,874 # ADC_LINE(0): 1
2021-11-11 14:29:13,874 # ADC_LINE(1): 2817
2021-11-11 14:29:13,974 # ADC_LINE(0): 2
2021-11-11 14:29:13,974 # ADC_LINE(1): 4095
2021-11-11 14:29:14,074 # ADC_LINE(0): 1
2021-11-11 14:29:14,074 # ADC_LINE(1): 4095
2021-11-11 14:29:14,174 # ADC_LINE(0): 2
2021-11-11 14:29:14,174 # ADC_LINE(1): 4094
2021-11-11 14:29:14,274 # ADC_LINE(0): 1
2021-11-11 14:29:14,274 # ADC_LINE(1): 4095
2021-11-11 14:29:14,374 # ADC_LINE(0): 1
2021-11-11 14:29:14,374 # ADC_LINE(1): 4095
2021-11-11 14:29:14,474 # ADC_LINE(0): 2
2021-11-11 14:29:14,474 # ADC_LINE(1): 4095
2021-11-11 14:29:14,574 # ADC_LINE(0): 0
2021-11-11 14:29:14,574 # ADC_LINE(1): 4095
```

### CODE REVIEW

File: boards/lora-e5-dev/Makefile.features [hunk 1]
```diff
@@ -5,6 +5,7 @@ CPU_MODEL = stm32wle5jc
 FEATURES_PROVIDED += periph_i2c
 FEATURES_PROVIDED += periph_lpuart
 FEATURES_PROVIDED += periph_rtt
+FEATURES_PROVIDED += periph_adc
```
Comment (aabadie) [inline boards/lora-e5-dev/Makefile.features:8]: Please keep the alphabetical order

File: cpu/stm32/periph/Kconfig [hunk 2]
```diff
@@ -13,10 +13,10 @@ config MODULE_PERIPH
     help
         stm32 common peripheral code.
 
-config MODULE_PERIPH_UART_NONBLOCKING
-    depends on HAS_PERIPH_UART_NONBLOCKING
-    depends on MODULE_PERIPH_UART
-    select MODULE_TSRB
+config module_periph_uart_nonblocking
```
Comment (aabadie) [inline cpu/stm32/periph/Kconfig:16]: Seems like an unrelated change

**Review by aabadie — CHANGES_REQUESTED** [review]
Some nits. Is this driver so different from the L4 version (I put on side the xtimer/ztimer use) ?

**Comment by fjmolinas** [comment]
> Some nits. Is this driver so different from the L4 version (I put on side the xtimer/ztimer use) ?

Well, there is only one ADC, clock source is not the same. It resembles f0/g0 more IMO, or is a mix between g0/f0 and l4.

> Some nits. Is this driver so different from the L4 version (I put on side the xtimer/ztimer use) ?

Should I use xtimer? I don't care if we migrate all of them later.

**Comment by fjmolinas** [comment]
> > Some nits. Is this driver so different from the L4 version (I put on side the xtimer/ztimer use) ?
> 
> Well, there is only one ADC, clock source is not the same. It resembles f0/g0 more IMO, or is a mix between g0/f0 and l4.

I don't really want to go into a common refactoring that would mean testing multiple hardware, and looking through multiple RM and datasheets.

File: boards/lora-e5-dev/Makefile.features (hunk 1 above)
Comment (aabadie) [reply boards/lora-e5-dev/Makefile.features:8]: still unaddressed

**Comment by aabadie** [comment]
> Should I use xtimer?

I don't think so. Better go with ztimer from the start, this will be one less thing to migrate (and test) after.

File: boards/lora-e5-dev/Makefile.features (hunk 1 above)
Comment (fjmolinas) [reply boards/lora-e5-dev/Makefile.features:8]: Hmm though I had... sorry

**Review by aabadie — APPROVED** [review]
Test output is provided and code changes look good. Please squash !

ACK

**Comment by benpicco** [comment]
looks like Kconfig still needs an update

**Comment by aabadie** [comment]
My ACK holds, feel free to merge once Murdock is green.

**Comment by fjmolinas** [comment]
Go!, thanks for the review!

**Pull request merged**